### PR TITLE
New board, SD card, restructuring.

### PR DESCRIPTION
**Pull request by elmo128**
Added ESP32cam board (no display), setup via config file on sd card, moved certain parts of code, eg. nvMem access to a dedicated class.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,9 @@
 .pio
 .vscode
+.vs
+*.sln
+*.vcxproj*
+x64
```
Comment (BitMaker-hub) [inline .gitignore:6]: I saw this files. are you using did you tried on x64 platform using visual studio?

File: README.md [hunk 2]
```diff
@@ -37,8 +37,11 @@ Every time an stratum job notification is received miner update its current work
 - 3D BOX [here](3d_files/)
 
 #### Current Supported Boards
-- TTGO T-Display S3 ([Aliexpress link](https://s.click.aliexpress.com/e/_Ddy7739))
-- ESP32-WROOM-32, ESP32-Devkit1.. ([Aliexpress link](https://s.click.aliexpress.com/e/_DCzlUiX))
+- TTGO T-Display S3 ([Aliexpress link*](https://s.click.aliexpress.com/e/_Ddy7739))
+- ESP32-WROOM-32, ESP32-Devkit1.. ([Aliexpress link*](https://s.click.aliexpress.com/e/_DCzlUiX))
+- ESP32-cam [Board Info](https://lastminuteengineers.com/getting-started-with-esp32-cam/)
+
+* affiliate link
```
Comment (BitMaker-hub) [inline README.md:44]: I agree on adding this, and board info for each one

**Comment by BitMaker-hub** [comment]
elmo, please can you rebase your project from dev branch and then remake this PR into dev Branch?
There are several conflicts

File: src/drivers/devices/device.h
Comment (BitMaker-hub) [inline src/drivers/devices/device.h:1]: please, do the changes from dev branch updated at last state. I understand your driver structure and everything is ok on PR but there are missing boards on your files.

Can you do so into dev Branch?

File: src/drivers/storage/nvMemory.h
Comment (BitMaker-hub) [inline src/drivers/storage/nvMemory.h:1]: Ok I like it a lot

**Review by BitMaker-hub — CHANGES_REQUESTED** [review]
Fantastic PR. 
Just as said, please rebase your project with last changes and make a new PR.
Try working with last state of dev branch. 

Unfortunatelly I can't merge until this is done. Thanks for this proof of work

File: .gitignore (hunk 1 above)
Comment (elmo128) [reply .gitignore:6]: I used vs to write the majority of the code. x64 was created when i tried to set it up for platformio. These are just remaining artefacts.

File: README.md (hunk 2 above)
Comment (elmo128) [reply README.md:44]: You could add to the donation section of the readme, something like ...or order your board by using one of the affiliate links above and even provide links to other marketplaces too. My goal was to link to a detailed description of the board, as it is not esp32-S3 based.

File: src/drivers/devices/device.h
Comment (elmo128) [reply src/drivers/devices/device.h:1]: Sure.

File: src/drivers/storage/nvMemory.h
Comment (elmo128) [reply src/drivers/storage/nvMemory.h:1]: thanks

**Comment by elmo128** [comment]
created new PR as requested.

**Pull request closed without merging**